scipy/spatial/distance.py

```diff
@@ -1818,23 +1836,6 @@ def num_obs_y(Y):
     return d
 
 
-def _row_norms(X):
-    norms = np.einsum('ij,ij->i', X, X, dtype=np.double)
-    return np.sqrt(norms, out=norms)
-
-
-def _cosine_cdist(XA, XB, dm):
```

Comment:
Have yo timed the new C version of `cosine_cdist` against this removed implementation? Because it uses `np.dot`, which most likely uses a multithreaded library, it's likely to be faster.